src/kbmod/standardizers/butler_standardizer.py

```diff
@@ -331,8 +331,16 @@ def toLayeredImage(self):
         psfs = self.standardizePSF()
 
         # guaranteed to exist, i.e. safe to access
-        mjds = meta["mjd"]
+        if isinstance(meta["mjd"], (list, tuple)):
+            mjds = meta["mjd"]
+        else:
+            mjds = (meta["mjd"] for e in self.processable)
+
         imgs = []
         for sci, var, mask, psf, t in zip(sciences, variances, masks, psfs, mjds):
-            imgs.append(LayeredImage(RawImage(sci), RawImage(var), RawImage(mask), t, psf))
+            # Converts nd array mask from bool to np.float32
+            mask = mask.astype(np.float32)
```

Comment:
Why are we explicitly casting the mask to float32? Is Eigen complaining to you? If so then we should address this in the `standardizeMaskImage` method.

Reply:
I'm not quite sure if there is a good reason tbh. I saw that you added it in this ["Fix bug in Fits Standardizers." commit](https://github.com/dirac-institute/kbmod/pull/334/commits/7c44e788718ba7e34b1d38e72dbd0b593071c5b6) but not sure what it specifically was trying to fix

We also currently spoof that the mask should be `np.int32` [here](https://github.com/dirac-institute/kbmod/blob/43218e4e9580e3cf33dfa13dd6eadc2fe0de9fc2/tests/utils/mock_fits.py#L270)

FITS standardizer is fairly general so seems fine to keep the type conversion here to me, but I can look a bit deeper into it if you want

Reply:
Though actually taking another look, in main we do still still get back to a float32, probably in the conversion to Eigen when we create the `RawImage`

I'm starting to feel that we should remove all the type conversions both in here and in
![Uploading Screenshot 2024-03-19 at 3.08.51 PM.png…]()
`standardizeMaskImage`